server/sonar-server/src/main/java/org/sonar/server/ui/ViewProxy.java

```diff
@@ -272,6 +274,24 @@ public boolean acceptsAvailableMeasures(String[] availableMeasures) {
     }
   }
 
+  public boolean isUserAuthorized() {
+    boolean authorized = userRoles.length == 0;
+    for (String userRole : getUserRoles()) {
+      authorized |= UserSession.get().hasGlobalPermission(userRole);
+    }
+    return authorized;
+  }
+
+  public boolean isUserAuthorized(ComponentDto component) {
+    boolean authorized = userRoles.length == 0;
+    for (String userRole : getUserRoles()) {
+      authorized |= (UserRole.VIEWER.equals(userRole)
+        || UserRole.USER.equals(userRole)
+        || UserSession.get().hasProjectPermissionByUuid(userRole, component.uuid()));
```

Comment:
This condition was translated directly from the equivalent Rails code. After validation with @bellingard I created http://jira.codehaus.org/browse/SONAR-6525 and will fix this before merging.